### ISSUE TITLE
Rewrite docs for "Custom Route Predicates"

### ISSUE THREAD

**Issue by hellp** [Pylons/pyramid#1161]
The section "Custom Route Predicates" in narr/urldispatch.rst still uses the (as of 1.5) deprecated method of adding route predicates via `add_route/add_view(custom_predicates=...)`. It should be rewritten to use the new `Configurator.add_view_predicate`/`Configurator.add_route_predicate`.

**Comment by stevepiercy** [comment]
I took a look, and I am too dense to understand the big words. I've unassigned myself from this issue.

That said, I did find the following reference that perhaps could replace the current section entirely?
https://docs.pylonsproject.org/projects/pyramid/en/latest/narr/hooks.html#registering-thirdparty-predicates

**Comment by stevepiercy** [comment]
Per [discussion in IRC](https://botbot.me/freenode/pyramid/2018-09-10/?msg=104212118&page=2), my guess is the proper direction.

**Comment by merwok** [comment]
I was about to report this, having found no deprecation notice in https://docs.pylonsproject.org/projects/pyramid/en/latest/narr/viewconfig.html#view-configuration-parameters (sorry can’t link to exact section — another possible improvement).  I may tackle it after #3520 

Seems related to #3347 but that ticket has no info or links to commits.

**Comment by stevepiercy** [comment]
#3347 is a duplicate of this issue.

**Comment by stevepiercy** [comment]
@merwok would you like to take on this issue?

**Comment by merwok** [comment]
Yes, in a week or two.

**Comment by merwok** [comment]
PR will be here tomorrow.
